charts/dscp-ipfs-v2/templates/statefulset.yaml

```diff
@@ -0,0 +1,320 @@
+              mountPath: {{ .Values.persistence.mountPath }}
+              {{- if .Values.persistence.subPath }}
+              subPath: {{ .Values.persistence.subPath }}
+              {{- end }}
+        {{- end }}
+        {{- if .Values.ipfs.initConfig.enabled }}
+        - name: init-ipfs-config
+          image: {{ include "dscp-ipfs.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.containerSecurityContext.enabled }}
+          securityContext: {{- omit .Values.containerSecurityContext "enabled" | toYaml | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.command "context" $) | nindent 12 }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
+          {{- else }}
+          {{ if .Values.ipfs.command }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.ipfs.command "context" $) | nindent 12 }}
+          {{- else }}
+          command:
+            - /bin/sh
+            - -c
+            - -e
+          {{- end }}
+          {{ if .Values.ipfs.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.ipfs.args "context" $) | nindent 12 }}
+          {{- else }}
+          args:
+            - |
+              [ ! -f "{{ .Values.persistence.mountPath }}/config" ] && {{ .Values.ipfs.binary }} init --profile server;
+              {{ .Values.ipfs.binary }} config Addresses.API /ip4/0.0.0.0/tcp/{{ .Values.apiService.ports.api | default 5001 }};
+              {{ .Values.ipfs.binary }} config --json Addresses.Swarm "[\"/ip4/0.0.0.0/tcp/{{ .Values.swarmService.ports.swarm }}\"]";
+              {{ .Values.ipfs.binary }} config --json Addresses.Announce "[\"/dns4/{{ include "common.names.fullname" . }}-swarm.{{ .Release.Namespace }}.svc.cluster.local/tcp/{{ .Values.swarmService.ports.swarm }}/\"]";
+              {{ .Values.ipfs.binary }} config Datastore.StorageMax {{ .Values.persistence.size | replace "Gi" "Gb" }};
+              {{ .Values.ipfs.binary }} config Routing.Type dht;
+              {{ .Values.ipfs.binary }} config --json Discovery.MDNS.Enabled {{ true | toJson | quote }};
+              {{ .Values.ipfs.binary }} config --json Swarm.AddrFilters {{ .Values.ipfs.swarmAddrFilters | toJson | quote }};
+              {{ .Values.ipfs.binary }} config --json API.HTTPHeaders.Access-Control-Allow-Origin "[\"http://0.0.0.0:{{ .Values.apiService.ports.api }}\", \"http://127.0.0.1:{{ .Values.apiService.ports.api }}\"]";
+              {{ .Values.ipfs.binary }} bootstrap rm --all{{ if .Values.ipfs.bootNodeAddress }};
+              if [ ! -z "$BOOT_NODE_ADDRESS" ]; then {{ .Values.command }} bootstrap add {{ .Values.ipfs.bootNodeAddress | toJson | quote }}; fi;{{ end }}
+          {{- end }}
+          {{- end }}
```

Comment:
Did a significant tidy-up of this, only requires a few envars to be passed through, most other bits are handled by the template as per best practices.